### PR TITLE
feat(hycu): init hycu µapp

### PR DESCRIPTION
**Pull request by darsene**
ref: MANAGER-14729

| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/hycu`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-14729
| License          | BSD 3-Clause

- [ ] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

New µapp for HYCU product (Backup HYCU for OVHcloud

### CODE REVIEW

**Review by rjamet-ovh — APPROVED** [review]

File: packages/manager/apps/hycu/src/pages/listing/Listing.page.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    if (columns && status === 'success' && flattenData?.length > 0) {
+      const newColumns = Object.keys(flattenData[0])
+        .filter((element) => element !== 'iam')
+        .map((element) => ({
+          id: element,
+          label: element,
+          type: dataType(flattenData[0][element]),
+          cell: (props: any) => {
+            const label = props[element] as string;
+            if (typeof label === 'string' || typeof label === 'number') {
```
Comment (tibs245) [inline packages/manager/apps/hycu/src/pages/listing/Listing.page.tsx:63]: As is `any` type with external element. We need protect call (?)

```suggestion
  useEffect(() => {
    if (columns && status === 'success' && flattenData?.length > 0) {
      const newColumns = Object.keys(flattenData[0])
        .filter((element) => element !== 'iam')
        .map((element) => ({
          id: element,
          label: element,
          type: dataType(flattenData[0][element]),
          cell: (props: any) => {
            const label = props?.[element] as unknown;
            if (typeof label === 'string' || typeof label === 'number') {
```

File: packages/manager/apps/hycu/src/pages/listing/Listing.page.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                    </OsdsLink>
+                  </DataGridTextCell>
+                );
+              return <DataGridTextCell>{label}</DataGridTextCell>;
+            }
+            return <div>-</div>;
+          },
+        }));
+      setColumns(newColumns);
+    } else if (status === 'success' && !flattenData?.length) {
```
Comment (tibs245) [inline packages/manager/apps/hycu/src/pages/listing/Listing.page.tsx:81]:
```suggestion
    } else if (isSuccess && !flattenData?.length) {
```

File: packages/manager/apps/hycu/src/pages/listing/Listing.page.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  });
+
+  const navigateToDashboard = (label: string) => {
+    const path =
+      location.pathname.indexOf('pci') > -1 ? `${location.pathname}/` : '/';
+    navigate(`${path}${label}`);
+  };
+
+  useEffect(() => {
+    if (columns && status === 'success' && flattenData?.length > 0) {
```
Comment (tibs245) [inline packages/manager/apps/hycu/src/pages/listing/Listing.page.tsx:54]:
```suggestion
    if (columns && isSuccess && flattenData?.length > 0) {
```

File: packages/manager/apps/hycu/src/pages/onboarding/Onboarding.page.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    {
+      id: 3,
+      texts: {
+        title: t('guide3Title'),
+        description: t('guide3Description'),
+        category: t('guideCategory'),
+      },
+      href: link?.guideLink3,
+    },
+  ];
```
Comment (tibs245) [inline packages/manager/apps/hycu/src/pages/onboarding/Onboarding.page.tsx:40]:
```suggestion
  const TILE_LIST = [
    {
      id: 1,
      texts: {
        title: t('guide1Title'),
        description: t('guide1Description'),
        category: t('guideCategory'),
      },
      href: link?.guideLink1,
    },
    {
      id: 2,
      texts: {
        title: t('guide2Title'),
        description: t('guide2Description'),
        category: t('guideCategory'),
      },
      href: link?.guideLink2,
    },
    {
      id: 3,
      texts: {
        title: t('guide3Title'),
        description: t('guide3Description'),
        category: t('guideCategory'),
      },
      href: link?.guideLink3,
    },
  ] as const;
```

File: packages/manager/apps/hycu/src/pages/onboarding/index.scss [hunk 5]
```diff
@@ -0,0 +1,10 @@
+.tile-grid {
+  display: grid;
+  grid-template-columns: repeat(2, 1fr);
+  grid-gap: 30px;
```
Comment (tibs245) [inline packages/manager/apps/hycu/src/pages/onboarding/index.scss:4]: Not rem for gap ?

File: packages/manager/apps/hycu/src/routes/routes.constant.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+export const urls = {
+  root: '/',
+  onboarding: '/onboarding',
+  listing: '/',
+  dashboard: '/:serviceName',
+  tab2: 'Tab2',
+};
```
Comment (tibs245) [inline packages/manager/apps/hycu/src/routes/routes.constant.ts:7]:
```suggestion
export const urls = {
  root: '/',
  onboarding: '/onboarding',
  listing: '/',
  dashboard: '/:serviceName',
  tab2: 'Tab2',
} as const;
```

File: packages/manager/apps/hycu/src/tracking.constant.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    config: {
+      level2: '98',
+    },
+  },
+  US: {
+    config: {
+      level2: '98',
+    },
+  },
+};
```
Comment (tibs245) [inline packages/manager/apps/hycu/src/tracking.constant.ts:17]:
```suggestion
export const LEVEL_2 = {
  EU: {
    config: {
      level2: '98',
    },
  },
  CA: {
    config: {
      level2: '98',
    },
  },
  US: {
    config: {
      level2: '98',
    },
  },
} as const;
```

**Review by tibs245 — COMMENTED** [review]
It's only supposition and comment to known if is relevant or not is OVH manage context. For a lot comment it's just details may be not relevant ;)

File: packages/manager/apps/hycu/src/hooks/guide/useGuideUtils.tsx [hunk 8]
```diff
@@ -0,0 +1,100 @@
+import { useContext, useEffect, useState } from 'react';
+import { CountryCode } from '@ovh-ux/manager-config';
+import { ShellContext } from '@ovh-ux/manager-react-shell-client';
+
+const docUrl = 'https://docs.ovh.com';
```
[the remark this answers is not in the document]
Comment (darsene) [reply packages/manager/apps/hycu/src/hooks/guide/useGuideUtils.tsx:5]: it's a mock list for now, so I'll update it later

File: packages/manager/apps/hycu/src/pages/onboarding/Onboarding.page.tsx (hunk 4 above)
Comment (darsene) [reply packages/manager/apps/hycu/src/pages/onboarding/Onboarding.page.tsx:40]: The onboarding page will be updated so I ignore it for now

File: packages/manager/apps/hycu/src/pages/listing/Listing.page.tsx (hunk 3 above)
Comment (darsene) [reply packages/manager/apps/hycu/src/pages/listing/Listing.page.tsx:54]: the hook `useResourcesIcebergV6` doesn't return the transtack query value, so for now I just store it on a const until it's provided later

File: packages/manager/apps/hycu/src/pages/onboarding/index.scss (hunk 5 above)
Comment (darsene) [reply packages/manager/apps/hycu/src/pages/onboarding/index.scss:4]: seems that file is not used. It should be removed from the generator rather

File: packages/manager/apps/hycu/src/pages/listing/Listing.page.tsx (hunk 1 above)
Comment (darsene) [reply packages/manager/apps/hycu/src/pages/listing/Listing.page.tsx:63]: will be removed later

File: packages/manager/apps/hycu/src/hooks/breadcrumb/useBreadcrumb.tsx [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    const fetchRoot = async () => {
+      try {
+        const response = await shell?.navigation.getURL(appName, '#/', {});
+        const rootItem = {
+          label: rootLabel,
+          href: String(response),
+        };
+        setRoot([rootItem]);
+      } catch {
+        // Fetch navigation error
```
[the remark this answers is not in the document]
Comment (darsene) [reply packages/manager/apps/hycu/src/hooks/breadcrumb/useBreadcrumb.tsx:34]: Will be managed after during the development

File: packages/manager/apps/hycu/src/tracking.constant.ts (hunk 7 above)
Comment (darsene) [reply packages/manager/apps/hycu/src/tracking.constant.ts]: I kept `LEVEL2` to be consistent with other apps

**Review by tibs245 — APPROVED** [review]

File: packages/manager/apps/hycu/src/pages/listing/Listing.page.tsx (hunk 3 above)
Comment (aboungnaseng-ovhcloud) [reply packages/manager/apps/hycu/src/pages/listing/Listing.page.tsx]: Yes we can return this property if needed 👍

File: packages/manager/apps/hycu/src/components/Loading/Loading.tsx [hunk 10]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { OsdsSpinner } from '@ovhcloud/ods-components/react';
+
+export default function Loading() {
+  return (
+    <div className="flex justify-center">
+      <div>
+        <OsdsSpinner />
+      </div>
+    </div>
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/manager/apps/hycu/src/components/Loading/Loading.tsx]: unecessary sure 👍

**Review by aboungnaseng-ovhcloud — APPROVED** [review]

**Pull request merged**